### PR TITLE
Tkey support

### PR DESCRIPTION
**Pull request by cmouse**
Make PowerDNS aware of TKEY record, needed for future GSS support. Depends on #2157

### CODE REVIEW

**Comment by cmouse** [comment]
Needs rebasing after #2157

**Comment by cmouse** [comment]
This Pull Request is ready for merge.

**Comment by cmouse** [comment]
Thank you

**Comment by cmouse** [comment]
@habbie should I squash or rebuild this after it has been fixed up? too many commits per changed lines

**Comment by Habbie** [comment]
Some squashing is fine, but please keep the commits sensible and small.

**Comment by cmouse** [comment]
ready for merge from on my part... review won't hurt

**Comment by cmouse** [comment]
removed.

**Comment by rubenk** [comment]
Looks good from my point of view. This might benefit from review from someone with more experience in the subject matter than me however.

File: regression-tests/tkey.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+import socket;
+import select;
+import sys;
```
Comment (Habbie) [inline regression-tests/tkey.py]: don't need ; here ;)

**Comment by Habbie** [comment]
Reviewed, ready to merge, but I won't do so before Thursday. Feel free to fix those semicolons :)

**Pull request merged**